File: lib/domain/repository/call_recording_repository.dart
import 'package:support_call_recorder/domain/model/call_recording.dart';
import 'package:support_call_recorder/domain/model/support_status.dart';

abstract class CallRecordingRepository {
  Future<List<CallRecording>> getAll();
  Future<CallRecording?> getById(int id);
  Future<int> insert(CallRecording recording);
  Future<void> updateNote(int id, String note);
  Future<void> updateStatus(int id, SupportStatus status);
}

File: app/src/main/java/com/reco/support/domain/repository/CallRecordingRepository.kt
package com.reco.support.domain.repository

import com.reco.support.domain.model.CallRecording
import com.reco.support.domain.model.SupportStatus
import kotlinx.coroutines.flow.Flow

interface CallRecordingRepository {
    fun observeAll(): Flow<List<CallRecording>>
    suspend fun getById(id: Long): CallRecording?
    suspend fun insert(recording: CallRecording): Long
    suspend fun updateNote(id: Long, note: String)
    suspend fun updateStatus(id: Long, status: SupportStatus)
}
